code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -0,0 +1,412 @@
+		return ..()
+	owner.adjustToxLoss(-2 * REM * creation_purity * delta_time, 0)
+	..()
+	return TRUE
+
+///Can bring a corpse back to life temporarily (if heart is intact)
+///Makes wounds bleed more, if it brought someone back, they take additional brute and heart damage
+///They can't die during this, but if they're past crit then take increasing stamina damage
+///If they're past fullcrit, their movement is slowed by half
+///If they OD, their heart explodes (if they were brought back from the dead)
+/datum/reagent/inverse/penthrite
+	name = "Nooartrium"
+	description = "A reagent that is known to stimulate the heart in a dead patient, temporarily bringing back recently dead patients at great cost to their heart."
+	ph = 14
+	metabolization_rate = 1 * REM
+	addiction_types = list(/datum/addiction/medicine = 12)
+	overdose_threshold = 20
+	///If we brought someone back from the dead
+	var/back_from_the_dead
+
+/datum/reagent/inverse/penthrite/on_mob_dead(mob/living/carbon/owner)
+	var/obj/item/organ/heart/heart = owner.getorganslot(ORGAN_SLOT_HEART)
+	if(!heart || heart.organ_flags & ORGAN_FAILING)
+		return ..()
+	ADD_TRAIT(owner, TRAIT_STABLEHEART, type)
+	ADD_TRAIT(owner, TRAIT_NOHARDCRIT, type)
+	ADD_TRAIT(owner, TRAIT_NOSOFTCRIT, type)
+	ADD_TRAIT(owner, TRAIT_NOCRITDAMAGE, type)
+	owner.stat = CONSCIOUS
+	back_from_the_dead = TRUE
+	owner.emote("gasp")
+
+/datum/reagent/inverse/penthrite/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	owner.playsound_local(owner, 'sound/health/slowbeat.ogg', 40)
+	if(back_from_the_dead)
+		owner.adjustBruteLoss(5 * (1-creation_purity) * delta_time)
+		owner.adjustOrganLoss(ORGAN_SLOT_HEART, 2.5 * (1-creation_purity) * delta_time)
+	for(var/datum/wound/iter_wound as anything in owner.all_wounds)
+		iter_wound.blood_flow += (1-creation_purity)
+	if(owner.health < HEALTH_THRESHOLD_CRIT)
+		owner.add_movespeed_modifier(/datum/movespeed_modifier/reagent/nooartrium)
+	if(owner.health < HEALTH_THRESHOLD_FULLCRIT)
+		owner.add_actionspeed_modifier(/datum/actionspeed_modifier/nooartium)
+
+/datum/reagent/inverse/penthrite/on_mob_delete(mob/living/carbon/owner)
+	REMOVE_TRAIT(owner, TRAIT_STABLEHEART, type)
+	if(owner.health < -500)//Honestly commendable
+		var/obj/item/organ/heart/heart = owner.getorganslot(ORGAN_SLOT_HEART)
+		explosion(owner, 1, 1, 1)
+		qdel(heart)
+		owner.visible_message("<span class='boldwarning'>[owner]'s heart explodes!</span>")
+	REMOVE_TRAIT(owner, TRAIT_NOHARDCRIT, type)
+	REMOVE_TRAIT(owner, TRAIT_NOSOFTCRIT, type)
+	REMOVE_TRAIT(owner, TRAIT_NOCRITDAMAGE, type)
+	owner.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/nooartrium)
+	owner.remove_actionspeed_modifier(/datum/actionspeed_modifier/nooartium)
+	. = ..()
+
+/datum/reagent/inverse/penthrite/overdose_start(mob/living/carbon/owner)
+	if(!back_from_the_dead)
+		return ..()
+	var/obj/item/organ/heart/heart = owner.getorganslot(ORGAN_SLOT_HEART)
```

Comment:
ok but what if their heart got removed

Reply:
Rewrote it a bit to deal with missing hearts.